plugins/utils/which-key.nix

```
@@ -55,257 +265,308 @@ with lib;
       ];
     };
 
+    notify = defaultNullOpts.mkBool true "Show a warning when issues were detected with your mappings.";
+
+    triggers = defaultNullOpts.mkListOf (with types; attrsOf anything) [
+      {
+        __unkeyed = "<auto>";
+        mode = "nxsot";
+      }
+    ] "Manually setup triggers";
+
+    defer =
+      defaultNullOpts.mkLuaFn
+        (mkRaw ''
+          function(ctx)
+            return ctx.mode == "V" or ctx.mode == "<C-V>
+            end'')
+        ''
+          Start hidden and wait for a key to be pressed before showing the popup.
+
+          Only used by enabled xo mapping modes.
+        '';
+
     plugins = {
-      marks = helpers.defaultNullOpts.mkBool true "shows a list of your marks on ' and `";
+      marks = defaultNullOpts.mkBool true "Shows a list of your marks on ' and `";
 
-      registers = helpers.defaultNullOpts.mkBool true ''shows your registers on " in NORMAL or <C-r> in INSERT mode'';
+      registers = defaultNullOpts.mkBool true ''Shows your registers on " in NORMAL or <C-r> in INSERT mode'';
 
       spelling = {
-        enabled = helpers.defaultNullOpts.mkBool true "enabling this will show WhichKey when pressing z= to select spelling suggestions";
-        suggestions = helpers.defaultNullOpts.mkInt 20 "how many suggestions should be shown in the list?";
+        enabled = defaultNullOpts.mkBool true "Enabling this will show WhichKey when pressing z= to select spelling suggestions";
+        suggestions = defaultNullOpts.mkInt 20 "How many suggestions should be shown in the list?";
       };
 
       presets = {
-        operators = helpers.defaultNullOpts.mkBool true "adds help for operators like d, y, ...";
-        motions = helpers.defaultNullOpts.mkBool true "adds help for motions";
-        textObjects = helpers.defaultNullOpts.mkBool true "help for text objects triggered after entering an operator";
-        windows = helpers.defaultNullOpts.mkBool true "default bindings on <c-w>";
-        nav = helpers.defaultNullOpts.mkBool true "misc bindings to work with windows";
-        z = helpers.defaultNullOpts.mkBool true "bindings for folds, spelling and others prefixed with z";
-        g = helpers.defaultNullOpts.mkBool true "bindings for prefixed with g";
+        operators = defaultNullOpts.mkBool true "Adds help for operators like d, y, ...";
+        motions = defaultNullOpts.mkBool true "Adds help for motions";
+        text_objects = defaultNullOpts.mkBool true "Help for text objects triggered after entering an operator";
+        windows = defaultNullOpts.mkBool true "Default bindings on <c-w>";
+        nav = defaultNullOpts.mkBool true "Misc bindings to work with windows";
+        z = defaultNullOpts.mkBool true "Bindings for folds, spelling and others prefixed with z";
+        g = defaultNullOpts.mkBool true "Bindings for prefixed with g";
```

These descriptions are probably short enough that they are better as they are instead of changing to indented strings.

Just need to add punctuation and markdown formatting.

```suggestion
        operators = defaultNullOpts.mkBool true "Adds help for operators like `d`, `y`, ...";
        motions = defaultNullOpts.mkBool true "Adds help for motions.";
        text_objects = defaultNullOpts.mkBool true "Help for text objects triggered after entering an operator.";
        windows = defaultNullOpts.mkBool true "Default bindings on `<c-w>`.";
        nav = defaultNullOpts.mkBool true "Misc bindings to work with windows.";
        z = defaultNullOpts.mkBool true "Show WhichKey for folds, spelling and other bindings prefixed with `z`.";
        g = defaultNullOpts.mkBool true "Show WhichKey for bindings prefixed with `g`.";
```